src/NHibernate/Event/Default/AbstractFlushingEventListener.cs

```diff
@@ -81,7 +81,7 @@ protected virtual void FlushEverythingToExecutions(FlushEvent @event)
 					.Append(" removals to ").Append(persistenceContext.CollectionEntries.Count).Append(" collections");
 
 				log.Debug(sb.ToString());
-				new Printer(session.Factory).ToString(persistenceContext.EntitiesByKey.Values.ToArray().GetEnumerator());
+				new Printer(session.Factory).ToString(persistenceContext.EntitiesByKey.Values.ToArray());
```

Comment:
Pattern change: take an `IEnumerable` instead of a disposable thing in "fire&forget" mode.